src/contracts/campaigns/BraidCampaign.sol

```diff
@@ -0,0 +1,195 @@
+pragma solidity ^0.4.4;
+
+
+import "utils/Owned.sol";
+import "verifiers/OwnedVerifier.sol";
+import "tokens/StandardToken.sol";
+
+/*
+A custom WeiFund compatable campaign for the Braid the Movie Ethereum campaign.
+*/
+
+contract BraidCampaign is Owned, StandardToken {
+  modifier pastFreezeDate {
+    // now must be greater than freeze date
+    if (block.number >= freezePeriod) {
+      _;
+    } else {
+      throw;
+    }
+  }
+
+  function transfer(address _to, uint256 _value) pastFreezeDate public returns (bool) {
+    return super.transfer(_to, _value);
+  }
+
+  function transferFrom(address _from, address _to, uint256 _value) pastFreezeDate public returns (bool success) {
+    return super.transferFrom(_from, _to, _value);
+  }
+
+  function approve(address _spender, uint256 _value) pastFreezeDate public returns (bool success) {
+    return super.approve(_spender, _value);
+  }
+
+  function () payable public {
+    // allow fallback function.. contribute as normal
+    contributeMsgValue(defaultAmounts);
+  }
+
+  function contributeMsgValue(uint256[] _amounts) payable public returns (uint256) {
+    // the total tokens amount to be issued to the sender
+    uint256 tokensOwed = msg.value / price;
```

Comment:
If we want more resolution, we can make tokens have 1e18 units, just like ether. (Let's call those units atoms to make it easier to talk about. One token has 1e18 atoms.) We need to go from wei to atoms without losing precision.

Currently, the price set by `payout-server` is in wei per token. If we want wei per atom, we need to divide the price by 1e18. If we do that, we're going to get zero—each token costs less than 1e18 wei, so integer division will discard the remainder. Instead, let's get the price in wei per 1e9 atoms. We're trying to avoid losing cents, and 1e9 atoms is one-ten millionth of a cent, so it's okay to lose precision there.

The price in wei per 1e9 atoms is `1 / usdPerEther * 1e18 / 1e9`. At $50 per ether, for instance, the price would be 2e7 wei per 1e9 atoms. In the contract, we'd take `msg.value / price * 1e9` to get the number of atoms the user should be issued for the ether they sent. A `decimals` attribute needs to be added to the contract and set to 18 so ERC20-compliant tools know to divide the value by 1e18 to display the number of tokens a user has.

TL;DR: Change `payout-server` to set the price as `const weiPricePerBillionth = oneEther.div(bodyData.mid).div('1e9');`, then change the contract to set `uint256 tokensOwed = msg.value / price * 1e9;` with a comment that reminds the reader that `price` is in wei per billionth of a token.